hail/src/main/scala/is/hail/variant/RegionValueVariant.scala

```diff
@@ -1,56 +1,54 @@
 package is.hail.variant
 
 import is.hail.annotations._
-import is.hail.expr.types.physical.{PArray, PString, PStruct}
+import is.hail.expr.types.physical.{PArray, PBinary, PLocus, PString, PStruct}
 import is.hail.utils._
 
 class RegionValueVariant(rowType: PStruct) extends View {
   private val locusField = rowType.fieldByName("locus")
+  private val locusPType = locusField.typ.asInstanceOf[PLocus]
   private val allelesField = rowType.fieldByName("alleles")
   private val locusIdx = locusField.index
   private val allelesIdx = allelesField.index
-  private val tl: PStruct = locusField.typ.fundamentalType.asInstanceOf[PStruct]
+  private val tl: PStruct = locusPType.fundamentalType.asInstanceOf[PStruct]
   private val taa: PArray = allelesField.typ.asInstanceOf[PArray]
-  private var region: Region = _
-  private var locusOffset: Long = _
+  val allelePType = taa.elementType.asInstanceOf[PString]
```

Comment:
Note that I don't assert the type before casting. Assert has some overhead, and to be completely accurate, we should be asserting all types that we're casting, which also means the variables that are not casted on master (`tl`, `taa`). Since casts were not previously used in this class, I left them off of my changes, and left this note to check whether we want to add all 4 necessary asserts.

In practice I believe all casts will fail (but want to check this), if an incompatible type is attempted. And since we're casting to base interfaces, that should mean this behavior is safe (if potentially more fragile).

Reply:
There is a larger question here as well: do we want to consistently assert before casting to PType interfaces (abstract classes)? Because if a cast to an incompatible PType can't succeed, we shouldn't incur the overhead, especially in interpreted code, but also in compilation. If however there are cases that a bad cast can succeed, or we want to enforce a strict policy on this for other casts where casting rules may be more subtle (I don't know of such cases, posing the question to you), we should enforce a codebase-wide policy on assertion before cast.